controllers/metal3machine_controller.go

```diff
@@ -248,6 +261,34 @@ func (r *Metal3MachineReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
+// ClusterToMetal3Machines is a handler.ToRequestsFunc to be used to enqeue
+// requests for reconciliation of Metal3Machines.
+func (r *Metal3MachineReconciler) ClusterToMetal3Machines(o handler.MapObject) []ctrl.Request {
+	result := []ctrl.Request{}
+	c, ok := o.Object.(*capi.Cluster)
+
+	if !ok {
+		r.Log.Error(errors.Errorf("expected a Cluster but got a %T", o.Object), "failed to get Metal3Machine for Cluster")
+		return nil
+	}
+
+	labels := map[string]string{capi.ClusterLabelName: c.Name}
+	capiMachineList := &capi.MachineList{}
+	if err := r.Client.List(context.TODO(), capiMachineList, client.InNamespace(c.Namespace), client.MatchingLabels(labels)); err != nil {
+		r.Log.Error(err, "failed to list Metal3Machines")
+		return nil
+	}
+	for _, m := range capiMachineList.Items {
+		if m.Spec.InfrastructureRef.Name == "" {
+			continue
+		}
+		name := client.ObjectKey{Namespace: m.Namespace, Name: m.Name}
```

Comment:
the namespace should come from the InfrastructureRef spec if provided

Reply:
I am not sure if I understood you correctly. I am following existing logic: https://github.com/metal3-io/cluster-api-provider-metal3/blob/master/controllers/metal3machine_controller.go#L281

Reply:
now you are assuming that the machine and Metal3Machine are in the same namespace. if the namespace is given in m.Spec.InfrastructureRef.Namespace, you should use that one, and default to the machine's namespace only if it is not given

Reply:
Is it possible to have machine and metal3lmachine in different namespaces ?

Reply:
Done